src/lib/ledgers/multiledger.js

```diff
@@ -1,38 +1,51 @@
 'use strict'
 
 const FiveBellsLedger = require('./five-bells-ledger')
+const healthStatus = require('../../common/health.js')
 
 function Multiledger (options) {
   this.config = options.config
   this.log = options.log
   this.ledger_types = {undefined: FiveBellsLedger}
   this.ledgers = {}
+  this.buildLedgers()
+  this.ledgersHealth = { ledgersHealth: healthStatus.statusNotOk }
 }
 
 Multiledger.prototype.addLedger = function (Ledger) {
   this.ledger_types[Ledger.TYPE] = Ledger
 }
 
 Multiledger.prototype.getLedger = function (ledgerId) {
-  return this.ledgers[ledgerId] ||
-        (this.ledgers[ledgerId] = this.buildLedger(ledgerId))
+  return this.ledgers[ledgerId]
 }
 
-Multiledger.prototype.buildLedger = function (ledgerId) {
-  let creds = this.config.getIn(['ledgerCredentials', ledgerId])
-  let Ledger = this.ledger_types[creds.type]
-  return new Ledger({
-    ledger_id: ledgerId,
-    credentials: creds,
-    log: this.log(Ledger.name),
-    config: this.config
+Multiledger.prototype.buildLedgers = function () {
+  Object.keys(this.config.get('ledgerCredentials')).map((ledgerId) => {
+    let creds = this.config.getIn(['ledgerCredentials', ledgerId])
+    let Ledger = this.ledger_types[creds.type]
+    this.ledgers[ledgerId] = new Ledger({
+      ledger_id: ledgerId,
+      credentials: creds,
+      log: this.log(Ledger.name),
+      config: this.config
+    })
   })
 }
 
 Multiledger.prototype.getType = function (ledgerId) {
   return this.getLedger(ledgerId).constructor.TYPE
 }
 
+Multiledger.prototype.checkLedgersHealth = function * () {
+  yield Object.keys(this.ledgers).map((ledgerId) => this.ledgers[ledgerId].checkHealth())
+  this.ledgersHealth = { ledgersHealth: healthStatus.statusOk }
```

Comment:
Once this is set to statusOk the first time, it can never be set to notOk because checkHealth will retrying with backoff forever. Is there ever a point where we might want to timeout?